[DOCS-1209][DOCS-1204] Document full-screen panel view, various panel sharing options, customizing number of panels per section (#1087)
wandb/docs: content/guides/models/app/features/panels/_index.md

1. At the top of the workspace, click the action menu `...`.
1. Click **Reset workspace**.

## View a panel in full-screen mode

In full-screen mode:

- A panel plots 10,000 buckets rather than 1,000 when not in full-screen mode.

> - The state of the run selector is preserved, so you can toggle runs on or off or search for runs.

Reviewer: I would just say "The run selector is still shown, so..."

> - The state of the run selector is preserved, so you can toggle runs on or off or search for runs.

Author: ```suggestion
- The run selector displays, so you can toggle runs on or off and search for runs.
```

1. At the top of the workspace, click the action menu `...`.
1. Click **Reset workspace**.

## View a panel in full-screen mode

> In full-screen mode:

Reviewer 2: This sentence and the beginning of the first bullet point sound a little repetitive (both start with "In full-screen").  Do we need bullet points here? Suggestion to condense this to:

"In full-screen mode, the run selector displays and panels use full full-fidelity sampling mode plots with 10,000 buckets. Non full-screen mode panels use 1,000 buckets , by default.

Author: Done

> - A panel plots 10,000 buckets rather than 1,000 when not in full-screen mode.

Reviewer 3: I don't think this is true in sampling mode, only in Full Fidelity

> - A panel plots 10,000 buckets rather than 1,000 when not in full-screen mode.

Author: ```suggestion
- In full-screen, a panel that uses full-fidelity sampling mode plots 10,000 buckets, rather than 1,000 when not in full-screen mode.
```

- The run selector displays, so you can toggle runs on or off and search for runs.

To view a panel in full-screen mode:

1. Hover over the panel.

> 1. Click the panel's action menu `...`, then click the full-screen button, which looks like a viewfinder.

Reviewer 3: There's a direct full screen button you can click on each panel w/out going into the menu

Author: True, but then you don't get the nice "Copy panel URL" UI prompt. Instead, you must copy it from the browser bar, or go into the resulting panel's three-dots menu to copy the URL at that point. So it seems like documenting the longer way is more general-purpose, and the shorter way is discoverable while you're following the instructions for the longer way. Does that make sense?

> 1. Click the panel's action menu `...`, then click the full-screen button, which looks like a viewfinder.

Reviewer 2: Wonder if we should replace "viewfinder" with something descriptive in case users don't know what that is? Something like, "...which looks like a viewfinder icon or the outline of a square..."

Author: Done, PTAL

To learn more about the options for each type of panel, refer to the relevant section below, such as [Line plots]({{< relref "line-plot/" >}}) or [Bar plots]({{< relref "bar-plot.md" >}}).

## Share panels

This section describes the various ways to share a panel and how each way differs.

The settings of a project determine who can view the panel. This means that if the project is private, only members of the project can view the panel. If the project is public, anyone with the link can view the panel.

> If multiple panels have the same name, W&B shares the first panel with the name.

Reviewer 4: This sentence feels a little out of place here. maybe nest it under "View and share a panel in full-screen mode", or create a new section along the lines of "Construct URL by panel name".

The format of the URL is:

```
https://wandb.ai/ENTITY_NAME/PROJECT_NAME?panelDisplayName=PANEL_NAME&panelSectionName=SECTON_NAME
```

Where `PANEL_NAME` is most commonly just the metric name, and the section name param is actually optional (but is one strategy for disambiguating if there are multiple panels with the same name across the workspace)

Knowing how to construct a panel URL is probably most relevant for those creating automations (e.g. user sets up a slack bot that will alert if a metric does something weird, and they want to include a link to the panel by the metric name)

Author: Done!

### Share a panel with a direct link

To share a panel with a specific user or team:

1. Hover over the panel, then click the panel's action menu `...`.
1. Click **Share panel in report**.
1. To invite a specific user by email, select the **Invite** tab.
    1. Enter an email address or username.
    1. Optionally, change **can view** to **can edit**.
    1. Click **Invite**. W&B sends an email to the user with a clickable link to the report.
1. To copy a shareable link to the panel, click the **Share** tab.
    1. Change **Only those who are invited have access** to **Anyone with the link can view**.
    1. Click **Copy report link**, then share the link with the user or team.
    1. Choose **Share on Twitter**, **Share on Reddit**, **Share on LinkedIn**, or **Copy embed link**.

When the user access the link, the panel opens in the normal view. You can instead [view and share a panel in full-screen mode]({{< relref "#view-and-share-a-panel-in-full-screen-mode" >}}).

> ### View and share a panel in full-screen mode

Reviewer: I would like this to be the primary action we guide people toward and I worry three things here are impeding that.  

First, we're conflating the full-screen with the direct-link-to-panel points.  Yes it's great that it's full screen, but the main improvement is that you are linking someone straight to the panel, rather than (as in the section titled "### Share a panel with a direct link") creating this pretty useless one-off Report that you then share.    I think the section should first be clear on the link sharing part.   The 10000 bins thing is a detail that fits better in the docs of the full screen panel mode itself than it does here in the sharing section.

Second, I think this section should come first, before "Share a panel with a direct link", which is the action we are basically obsoleting with this feature.

Third, I think we should rename the section currently called "Share a panel with a direct link", to something like what's in the product ("Share panel in report")  so that it's not as easily confused with the new feature.    In this action, you actually are not sharing the panel with a direct link, you are secretly creating a report, and sharing the link to **that**.

Author: Done, PTAL!

To learn more about the options for each type of panel, refer to the relevant section below, such as [Line plots]({{< relref "line-plot/" >}}) or [Bar plots]({{< relref "bar-plot.md" >}}).

## Share a panel

> This section shows how to share a panel. You can also [email a snapshot of a panel]({{< relref "#email-a-snapshot-of-a-panel" >}}).

Reviewer: I would not mention this so prominently or maybe even at all.  We're trying to explain and activate them on the most convenient and powerful method which is copying the panel url

> The settings of a project determine who can view the panel. This means that if the project is private, only members of the project can view the panel. If the project is public, anyone with the link can view the panel.

Reviewer: This true but also in practice not very relevant since practically all customer projects are private.  Could we leave this for lower down?

Author: Yeah it was really needed for explaining why you might not see the social links. Let me move / rework it.

{{% alert %}}
If you share a panel while viewing it in full-screen mode, the link opens the panel in full-screen mode. In full-screen mode:

- A panel plots 10,000 buckets rather than 1,000 when not in full-screen mode.

> - The state of the run selector is preserved, so you can toggle runs on or off or search for runs.

Reviewer: Here's a case where documenting full screen mode and sharing together confused me at least.  I initially thought you were implying that **sharing** the link preserves the run selector state, which it does not, I think.

{{% /alert %}}

To view and share a panel in full-screen mode:

1.  Complete the next steps from this menu.

> 1. **Optional**: To share the panel in full-screen mode, first view the panel in full-screen mode. Hover over the panel, click the panel's action menu `...`, then click the full-screen button, which looks like a viewfinder.

Reviewer: I don't think this is true.  When you share the panel URL from the normal multi-panel workspace view, it opens in full screen.

Generally I don't know why we are documenting full screen mode and sharing in the same section, rather than documented full screen mode in its own panel actions setting unrelated to sharing.   You can full screen without sharing.

1. Hover over the panel, then click the panel's action menu `...`.
1. Click the full-screen button, which looks like a viewfinder.
    {{< img src="/images/app_ui/panel_fullscreen.png" alt="View panel full-screen" >}}

    The panel opens in full-screen mode.
1. Copy the URL from the browser or click the action menu `...` and select **Copy panel URL**.
1. Share the link with the user or team. When they access the link, the panel opens in full-screen mode automatically.

The proceeding section shows how to determine a panel's full-screen URL programmatically.

#### Compose a panel's full-screen URL programmatically
In certain situations, such as when [creating an automation]({{< relref "/guides/models/automations/" >}}), it can be useful to include the panel's full-screen URL. This section shows how to compose The URL for a panel programmatically. In the proceeding example, replace the entity, project, panel, and section names in brackets.

```text
https://wandb.ai/<ENTITY_NAME>/<PROJECT_NAME>?panelDisplayName=<PANEL_NAME>&panelSectionName=<SECTON_NAME>
```

> If multiple panels have the same name, this is the URL for the first panel with the name. In this case, find the URL by [viewing the panel in full-screen mode]({{< relref "#view-and-share-a-panel-in-full-screen-mode" >}}).

Reviewer 4: ```suggestion
If multiple panels match the specified panel and section names, the first match will be selected.
```

I hesitate to point users back to the prev section, because the URL you get from "Copy panel URL" in the UI may still have the same problem (uses panel/section name query params that have "choose first" conflict resolution)

> ### Share a panel with a direct link that anyone can access

Reviewer: I feel like this and the following section are basically the same thing, the difference is whether it's a public link or not.   Maybe they could be consolidated?

1. Hover over the panel, then click the panel's action menu `...`.
1. Click **Share panel in report**.
1. At the bottom of the **Invite** tab, optionally change **Only those who are invited have access** to  **Anyone with the link can view**, then click **Copy report link**.
1. Share the link with the user or team.

### Share a panel as a report
1. Hover over the panel, then click the panel's action menu `...`.
1. Click **Share panel in report**.
1. In the **Invite** tab, enter an email address or username.
1. Specify **can view** or **can edit**.
1. Click **Invite**. W&B sends an email to the user with a clickable link to the report.

### Share or embed a panel publicly
To share a panel _publicly_, such as on social media or embedded in a website:
1. Hover over the panel, then click the panel's action menu `...`.
1. Click **Share panel in report**.
1. In the **Share** tab, change **Only those who are invited have access** to **Anyone with the link cah view**, then choose **Share on Twitter**, **Share on Reddit**, **Share on LinkedIn**, or **Copy embed link**.

> ### Determine a panel's full-screen URL programmatically

Reviewer: I think is hanging...you covered in a subsection above I think.

Reviewer: Maybe I'm misunderstanding.  I think it _is_ useful information we should documented, I just thought it was documented in the section "Compose a panel's full-screen URL programmatically" above.

Author: Yeah I think I finally understood, and have tried to address this in the latest commit.

To learn more about the options for each type of panel, refer to the relevant section below, such as [Line plots]({{< relref "line-plot/" >}}) or [Bar plots]({{< relref "bar-plot.md" >}}).

## Share a panel

This section shows how to share a panel using a link.

To share a panel, you can either:

- While viewing the panel in full-screen mode, copy the URL from the browser.
- Click the action menu `...` and select **Copy panel URL**.

> Share the link with the user or team. When they access the link, tpanel opens in [full-screen mode]({{< relref "#view-a-panel-in-full-screen-mode" >}}).

Reviewer: typo

> Share the link with the user or team. When they access the link, tpanel opens in [full-screen mode]({{< relref "#view-a-panel-in-full-screen-mode" >}}).

Author: ```suggestion
Share the link with the user or team. When they access the link, the panel opens in [full-screen mode]({{< relref "#view-a-panel-in-full-screen-mode" >}}).
```

```text
https://wandb.ai/<ENTITY_NAME>/<PROJECT_NAME>?panelDisplayName=<PANEL_NAME>&panelSectionName=<SECTON_NAME>
```

If multiple panels in the same section have the same name, this URL opens the first panel with the name.

### Embed or share a panel on social media
To embed a panel in a website or share it on social media, the panel must be viewable by anyone with the link. If a project is private, only members of the project can view the panel. If the project is public, anyone with the link can view the panel.

To get the code to embed or share a panel on social media:

1. From the workspace, hover over the panel, then click its action menu `...`.
1. Click the **Share** tab.
1. Change **Only those who are invited have access** to **Anyone with the link can view**. Otherwise, the choices in the next step are not available.
1. Choose **Share on Twitter**, **Share on Reddit**, **Share on LinkedIn**, or **Copy embed link**.

### Email a snapshot of a panel

> This section shows how to email a point-in-time snapshot of a panel as a report. After you share it, the report does not update. By contrast, when you [share a panel]({{< relref "#share-a-panel" >}}).

Reviewer: The third sentence I think cuts off.  But I also don't think this is true that the plot in the report remains a point in time, rather than continuing to update as the run updates?

Author: PTAL!

To learn more about the options for each type of panel, refer to the relevant section below, such as [Line plots]({{< relref "line-plot/" >}}) or [Bar plots]({{< relref "bar-plot.md" >}}).

## Share a panel

This section shows how to share a panel using a link.

> To share a panel, you can either:

Reviewer 2: ```suggestion
To share a panel using a link, you can either:
```

Author: Done

- While viewing the panel in full-screen mode, copy the URL from the browser.
- Click the action menu `...` and select **Copy panel URL**.

Share the link with the user or team. When they access the link, the panel opens in [full-screen mode]({{< relref "#view-a-panel-in-full-screen-mode" >}}).

> To get back to a panel's workspace while viewing a panel in full-screen mode, click the left-pointing arrow at the top of the page.

Reviewer 2: ```suggestion
To return to a panel's workspace while viewing a panel in full-screen mode, click the left-pointing arrow at the top of the page.
```

Author: Done

```text
https://wandb.ai/<ENTITY_NAME>/<PROJECT_NAME>?panelDisplayName=<PANEL_NAME>&panelSectionName=<SECTON_NAME>
```

If multiple panels in the same section have the same name, this URL opens the first panel with the name.

### Embed or share a panel on social media
To embed a panel in a website or share it on social media, the panel must be viewable by anyone with the link. If a project is private, only members of the project can view the panel. If the project is public, anyone with the link can view the panel.

To get the code to embed or share a panel on social media:

1. From the workspace, hover over the panel, then click its action menu `...`.
1. Click the **Share** tab.
1. Change **Only those who are invited have access** to **Anyone with the link can view**. Otherwise, the choices in the next step are not available.
1. Choose **Share on Twitter**, **Share on Reddit**, **Share on LinkedIn**, or **Copy embed link**.

### Email a panel report

> This section shows how to email a single panel as a stand-alone report.

Reviewer 2: ```suggestion
To email a single panel as a stand-alone report:
```

Author: Done

To add a new section before or after an existing section, you can instead click the section's action `...` menu, then click **New section below** or **New section above**.

### Manage a section's visible panels
By default, each section shows 6 panels. To customize a section that has more than 6 panels:

1. At the top of the section, click **1 to 6 of <X>**, where `<X>` is the total number of panels.

> 1. Choose how many panels to show, or click **Show all panels**.

Reviewer 4: i believe we enforce a max page size of 100 panels. "show all" will only be available if num panels in section < that max

i don't think it's necessary to explicitly call out the "show all" option.

```suggestion
1. Choose how many panels to show per page.
```

Author: Done!

To add a new section before or after an existing section, you can instead click the section's action `...` menu, then click **New section below** or **New section above**.

### Manage a section's visible panels

> By default, each section shows 6 panels, but you can show a maximum of 100 panels. To customize a section that has more than 6 panels:

Reviewer: I don't think this is true?  The default number depends on the width of the panels

> ### Manage a section's visible panels
> By default, each section shows 6 panels, but you can show a maximum of 100 panels. To customize a section that has more than 6 panels:
> 
> 1. At the top of the section, click **1 to 6 of <X>**, where `<X>` is the total number of panels.
> 1. Choose how many panels to show.

Author: ```suggestion
### Manage a section's panels
If a section has a large number of panels, by default they are paginated.

- If a section uses the **Standard grid**, you can customize the number of panels to show:
  1. At the top of the section, click **1 to <X> of <Y>**, where `<X>` is the number of visible panels and `<Y>` is the total number of panels.
  1. Choose how many panels to show, up to 100..
- To show all panels when there are a large number of them, click the section's action `...` menu, then click **Custom grid**.
- To resize a panel, hover over it and use the drag handle to resize it.
  - If the section uses the **Standard grid**, all of the section's panels are updated to the new size.
  - If the section uses the **Custom grid**, only the panel you are resizing is updated.
- To delete a panel from a section:
  1. Hover over the panel, then click its action `...` menu.
  1. Click **Delete**.
  
  If you reset a workspace to an automated workspace, all deleted panels appear again.
```

To add a new section before or after an existing section, you can instead click the section's action `...` menu, then click **New section below** or **New section above**.

### Manage a section's panels

> If a section has a large number of panels, by default they are paginated.

Reviewer 2: ```suggestion
Sections with a large number of panels are paginated by default.
```

Author: Done

> If a section has a large number of panels, by default they are paginated.

Reviewer 2: Can we find out how many panels and replace "large number"?

Author: It depends on the panel width, and each panel can be resized separately.

Author: Done. PTAL

> - If a section uses the **Standard grid**, you can customize the number of panels to show:

Reviewer 2: I'm looking at the UI and I don't see a UI element called **Standard grid** in a project workspace. Is it a UI element or a description of a type of panel layout?

If it's not a UI element, suggest replacing with italicized  and adding a definition. Same applies to Custom grid.

Author: Done, PTAL
